src/vanilla/utils/selectAtom.ts

```diff
@@ -29,6 +31,12 @@ export function selectAtom<Value, Slice>(
   return memo3(
     () => {
       const EMPTY = Symbol()
+      const initState = () => ({
+        version: 0,
+        prev: EMPTY as Slice | typeof EMPTY | Promise<Slice>,
+      })
+      const ref = getCached(initState, stateCache, anAtom)
```

Comment:
This is shared across different Providers/stores.

Reply:
resolves in a follow-up commit, but then replaced by a commit that removes this approach entirely.